packages/tenant-process/src/services/tenantService.ts

```diff
@@ -486,6 +489,54 @@ export function tenantServiceBuilder(
       );
     },
 
+    async addTenantMail(
+      {
+        tenantId,
+        mailSeed,
+        organizationId,
+        correlationId,
+      }: {
+        tenantId: TenantId;
+        mailSeed: tenantApi.MailSeed;
+        organizationId: TenantId;
+        correlationId: string;
+      },
+      logger: Logger
+    ): Promise<void> {
+      logger.info(`Adding mail of kind ${mailSeed.kind} to Tenant ${tenantId}`);
+
+      await assertRequesterAllowed(tenantId, organizationId);
+
+      const tenant = await retrieveTenant(tenantId, readModelService);
+
+      if (tenant.data.mails.find((m) => m.address === mailSeed.address)) {
```

Comment:
This "find" should be working on ID if [this](https://github.com/pagopa/interop-be-monorepo/pull/466#discussion_r1694836722) comment will be confirmed.